conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt

```diff
@@ -0,0 +1,515 @@
+    ) {
+        if (action != Action.ADD && action != Action.REMOVE && action != Action.SET) {
+            logger.error("Action $action passed through to updateTablePermissions is unhandled. Doing no operations")
+            return
+        }
+
+        val allOrgs = columnsById.values.map { it.organizationId }
+        val removes = mutableMapOf<UUID, MutableList<String>>()
+        val adds = mutableMapOf<UUID, MutableList<String>>()
+        columnAcls.forEach { columnAcl ->
+            // Note - only handling propertyType aclKeys in here to add permissions to data
+            val column = columnsById.getValue(columnAcl.aclKey)
+            val orgId = column.organizationId
+            columnAcl.aces.forEach { ace ->
+                when (action) {
+                    Action.ADD -> {
+                        val sqls = adds.getOrDefault(orgId, mutableListOf())
+                        sqls.addAll( updatePermissionsOnColumnSql(
+                                ace,
+                                column,
+                                PgPermAction.GRANT
+                        ) )
+                        adds.set( orgId, sqls)
+                    }
+                    Action.REMOVE -> {
+                        val sqls = removes.getOrDefault(orgId, mutableListOf())
+                        sqls.addAll(
+                                removeAllPermissionsForPrincipalOnColumn(
+                                        ace.principal, column, tableType
+                                )
+                        )
+                        removes.set( orgId, sqls)
+                    }
+                    Action.SET -> {
+                        val remSqls = removes.getOrDefault(orgId, mutableListOf())
+                        remSqls.addAll(
+                                removeAllPermissionsForPrincipalOnColumn(
+                                        ace.principal, column, tableType
+                                )
+                        )
+                        removes.set( orgId, remSqls)
+
+                        val addSqls = adds.getOrDefault(orgId, mutableListOf())
+                        addSqls.addAll(  updatePermissionsOnColumnSql(
+                                ace,
+                                column,
+                                PgPermAction.GRANT
+                        ))
+                        adds.set( orgId, addSqls)
+                    }
+                }
+            }
+        }
+
+        allOrgs.forEach { organizationId ->
+            val rems = removes.getOrDefault(organizationId, listOf<String>())
+            val addz = adds.getOrDefault(organizationId, listOf<String>())
+            extDbManager.connectToOrg(organizationId).connection.use { conn ->
+                conn.autoCommit = false
+                val stmt: Statement = conn.createStatement()
+                try {
+                    rems.forEach {
+                        stmt.execute(it)
```

Comment:
why not do the rems as a batch too?

Reply:
wasn't sure about ordering/multiple executeBatch() calls on one statement object